pallets/relayer/src/lib.rs

```diff
@@ -0,0 +1,303 @@
+        /// The `user_key` is not attached to a CDD'd identity.
+        PayingKeyCddMissing,
+        /// The `user_key` already has a `paying_key`.
+        AlreadyHasPayingKey,
+        /// The `user_key` doesn't have a `paying_key`.
+        NoPayingKey,
+        /// The `user_key` has a different `paying_key`.
+        NotPayingKey,
+        /// The signer not authorized for `paying_key`.
+        NotAuthorizedForPayingKey,
+        /// The signer not authorized for `user_key`.
+        NotAuthorizedForUserKey,
+    }
+}
+
+impl<T: Config> Module<T> {
+    fn base_set_paying_key(origin: T::Origin, user_key: T::AccountId) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: paying_key,
+            primary_did: paying_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // Create authorization for setting the `paying_key` to the `user_key`
+        Self::unsafe_add_auth_for_paying_key(paying_did, user_key, paying_key, 0u128.into());
+        Ok(())
+    }
+
+    fn base_accept_paying_key(origin: T::Origin, auth_id: u64) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender: user_key, ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+        let signer = Signatory::Account(user_key);
+
+        <Identity<T>>::accept_auth_with(&signer, auth_id, |data, auth_by| -> DispatchResult {
+            let (user_key, paying_key, polyx_limit) = match data {
+                AuthorizationData::AddRelayerPayingKey(user_key, paying_key, polyx_limit) => {
+                    Ok((user_key, paying_key, polyx_limit))
+                }
+                _ => Err(AuthorizationError::BadAuthType),
+            }?;
+
+            Self::auth_accept_paying_key(signer.clone(), auth_by, user_key, paying_key, polyx_limit)
+        })
+    }
+
+    fn base_remove_paying_key(
+        origin: T::Origin,
+        user_key: T::AccountId,
+        paying_key: T::AccountId,
+    ) -> DispatchResult {
+        let PermissionedCallOriginData {
+            sender,
+            primary_did: sender_did,
+            ..
+        } = <Identity<T>>::ensure_origin_call_permissions(origin)?;
+
+        // allow: `origin == user_key` or `origin == paying_key`
+        if sender != user_key && sender != paying_key {
+            // allow: `origin == primary key of user_key's identity`
+            ensure!(
+                <Identity<T>>::get_identity(&user_key) == Some(sender_did),
+                Error::<T>::NotAuthorizedForUserKey
+            );
+        }
+
+        // Check if there is a paying key.
+        ensure!(
+            <Subsidies<T>>::contains_key(&user_key),
+            Error::<T>::NoPayingKey
+        );
+
+        // Get current Subsidy
+        let subsidy = <Subsidies<T>>::get(&user_key);
+
+        // Check if the current paying key matches.
+        ensure!(subsidy.paying_key == paying_key, Error::<T>::NotPayingKey);
```

Comment:
This is duplicated below, so let's extract to a helper function.

Also, let's use the `Option<_>` I noted above to do:
```suggestion
        // Get current subsidy.
        let subsidy = match <Subsidies<T>>::get(&user_key) {
            // There was no subsidy.
            None => fail!(Error::<T>::NoPayingKey),
            // There is one, but the paying key was wrong.
            Some(s) if s.paying_key != paying_key => fail!(Error::<T>::NotPayingKey),
            Some(s) => s,
        };
```